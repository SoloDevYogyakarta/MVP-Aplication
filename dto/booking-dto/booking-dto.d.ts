export declare class BookingField {
    name: string;
    title: string;
    desc: string;
    date: Date;
    time: string;
    user_id?: number;
}
