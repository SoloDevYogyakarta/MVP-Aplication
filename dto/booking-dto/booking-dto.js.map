{"version":3,"file":"booking-dto.js","sourceRoot":"","sources":["../../../src/dto/booking-dto/booking-dto.ts"],"names":[],"mappings":";;;;;;;;;;;;AAAA,qDAAyE;AAEzE,MAAa,YAAY;CAqBxB;AArBD,oCAqBC;AAlBC;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;0CACA;AAGb;IADC,IAAA,0BAAQ,GAAE;;2CACG;AAId;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;0CACA;AAGb;IADC,IAAA,wBAAM,GAAE;8BACH,IAAI;0CAAC;AAGX;IADC,IAAA,0BAAQ,GAAE;;0CACE;AAIb;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;6CACI"}