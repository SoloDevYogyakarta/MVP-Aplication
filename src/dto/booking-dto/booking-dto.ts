import { IsDate, IsNumber, IsOptional, IsString } from 'class-validator';

export class BookingField {
  @IsString()
  @IsOptional()
  name: string;

  @IsString()
  title: string;

  @IsString()
  @IsOptional()
  desc: string;

  @IsDate()
  date: Date;

  @IsString()
  time: string;

  @IsNumber()
  @IsOptional()
  user_id?: number;
}
