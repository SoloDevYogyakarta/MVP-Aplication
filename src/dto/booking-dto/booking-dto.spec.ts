import { faker } from '@faker-js/faker';
import { validate } from 'class-validator';
import moment from 'moment';
import { BookingField } from './booking-dto';

describe('BookingDto', () => {
  it('should to be defined', () => expect(BookingField).toBeDefined());

  it('render correctly', () => expect(BookingField).toMatchSnapshot());

  it('should to be successfully', async () => {
    const data = new BookingField();
    data.name = faker.person.fullName();
    data.title = faker.lorem.paragraph();
    data.desc = faker.lorem.paragraph();
    data.date = new Date();
    data.time = moment(new Date()).format('HH:mm:ss');
    const valid = await validate(data);
    expect(valid.length).toEqual(0);
  });
});
