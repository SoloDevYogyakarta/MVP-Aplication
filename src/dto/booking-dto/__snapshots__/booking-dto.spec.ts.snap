// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingDto render correctly 1`] = `[Function]`;
